portal/account/register.php

```diff
@@ -0,0 +1,556 @@
+            }
+        });
+    });
+
+    $('#selLanguage').on('change', function () {
+        callServer("set_lang", this.value);
+    });
+
+    $(document.body).on('hidden.bs.modal', function () { //@TODO maybe make a promise for wiz exit
+        callServer('cleanup');
+    });
+
+});
+
+function doCredentials(pid) { //@TODO Could get rid of and call callServer direct.
+    callServer('do_signup', pid);
+}
+
+function checkRegistration(pid) {
+    var profile = $("#profileFrame").contents();
+    var curStep = $("#step-2"),
+    curStepBtn = curStep.attr("id"),
+    nextstepwiz = $('div.setup-panel div a[href="#' + curStepBtn + '"]').parent().next().children("a"),
+    curInputs = $("#profileFrame").contents().find("input[type='text'],input[type='email'],select"),
+    isValid = true;
+    $(".form-group").removeClass("has-error");
+    var flg = 0;
+    for (var i = 0; i < curInputs.length; i++) {
+        if (!curInputs[i].validity.valid) {
+            isValid = false;
+            if (!flg) {
+                curInputs[i].scrollIntoView();
+                curInputs[i].focus();
+                flg = 1;
+            }
+            $(curInputs[i]).closest(".form-group").addClass("has-error");
+        }
+    }
+
+    if (!isValid) {
+        return false;
+    }
+
+    return true;
+}
+
+function callServer(action, value, value2, last, first) {
+    var data = {
+        'action' : action,
+        'value' : value,
+        'dob' : value2,
+        'last' : last,
+        'first' : first
+    }
+    if (action == 'do_signup') {
+        data = {
+            'action': action,
+            'pid': value
+        };
+    }
+    if (action == 'notify_admin') {
+        data = {
+            'action': action,
+            'pid': value,
+            'provider': value2
+        };
+    }
+    if (action == 'cleanup') {
+        data = {
+            'action': action
+        };
+    }
+    // The magic that is jquery ajax.l
+    $.ajax({
+        type : 'GET',
+        url : 'account.lib.php',
+        data : data
+    }).done(function (rtn) {
+        if (action == "cleanup") {
+            window.location.href = "./../index.php" // Goto landing page.
+        }
+        if (action == "set_lang") {
+            window.location.href = window.location.href;
+        }
+        if (action == "get_newpid") {
+            if (parseInt(rtn) > 0) { // set pid for restfull save
+                newPid = rtn;
+                $("#profileFrame").contents().find('input#pubpid').val(newPid);
+                $("#profileFrame").contents().find('input#pid').val(newPid);
+            } else { // After error alert app exit to landing page.
+                eModal.alert(rtn); // Existing user error. Error message is translated in account.lib.php. :)
+            }
+        }
+        if (action == 'do_signup') {
+            if (rtn == "") {
+                var message = "<?php echo xlt('Unable to either create credentials or send email.'); ?>";
```

Comment:
when using a js literal (since it is passed to alert), should use xls instead of xlt. Note xls is basically addslashes(xl()) .